source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,2685 @@
+  entity or event
+- Changing an existing field from optional to required
+- Introducing a new, required field in the test format
+- Significant changes to test file execution (not BC)
+
+Backwards-compatible changes SHALL warrant a new minor version. These changes
+include, but are not limited to:
+
+- Additive changes, such as a introducing a new `Special Test Operation`_ or
+  type of supported entity or event
+- Changing an existing field from required to optional
+- Introducing a new, optional field in the test format
+- Minor changes to test file execution (BC)
+
+Small fixes and internal spec changes (e.g. grammar, adding clarifying text to
+the spec) MAY warrant a new patch version; however, patch versions SHOULD NOT
+alter the structure of the test format and thus SHOULD NOT be relevant to test
+files (as noted in `schemaVersion`_).
+
+
+Entity Map
+----------
+
+The entity map indexes arbitrary objects and values by unique names, so that
+they can be referenced from test constructs (e.g.
+`operation.object <operation_object_>`_). To ensure each test is executed in
+isolation, test runners MUST NOT share entity maps between tests. Most entities
+will be driver objects created by the `createEntities`_ directive during test
+setup, but the entity map may also be modified during test execution via the
+`operation.saveResultAsEntity <operation_saveResultAsEntity_>`_ directive.
+
+Test runners MAY choose to implement the entity map in a fashion most suited to
+their language, but implementations MUST enforce both uniqueness of entity names
+and referential integrity when fetching an entity. Test runners MUST raise an
+error if an attempt is made to store an entity with a name that already exists
+in the map and MUST raise an error if an entity is not found for a name or is
+found but has an unexpected type.
+
+Consider the following examples::
+
+    # Error due to a duplicate name (client0 was already defined)
+    createEntities:
+      - client: { id: client0 }
+      - client: { id: client0 }
+
+    # Error due to a missing entity (client1 is undefined)
+    createEntities:
+      - client: { id: client0 }
+      - session: { id: session0, client: client1 }
+
+    # Error due to an unexpected entity type (session instead of client)
+    createEntities:
+      - client: { id: client0 }
+      - session: { id: session0, client: client0 }
+      - session: { id: session1, client: session0 }
+
+
+Supported Entity Types
+~~~~~~~~~~~~~~~~~~~~~~
+
+Test runners MUST support the following types of entities:
+
+- MongoClient. See `entity_client`_ and `client`_.
+- Database. See `entity_database`_ and `database`_.
+- Collection. See `entity_collection`_ and `collection`_
+- ClientSession. See `entity_session`_ and `session`_.
+- GridFS Bucket. See `entity_bucket`_ and `bucket`_.
+- GridFS Stream. See `entity_stream`_.
+- ChangeStream. See `changeStream`_.
+- The following BSON types:
+
+  - 0x01 - 0x13
```

Comment:
This range includes some deprecated types but I don't see the need to exclude those. The only way we'd get one of those types into the entity map is if it was returned from a driver's API method and used with `saveResultAsEntity` -- so my understanding is that a driver would already need to support the type before it could reach this point.

The only reason this sub-list exists is to ensure that we bump versions if/when a _new_ BSON type is introduced and needs to be added as a possible entity type.

Reply:
For clarity I suggest adding this to the spec text, i.e.

- All known BSON types. For the present version of the spec, the following BSON types are known: 0x01-0x13, 0x7f, 0xff. Adding new BSON types to this list requires a (major|minor) version bump of this spec.

Reply:
Revised this bullet in most recent commit. Please review and resolve.